lib/services/database/comment/user_comment_repository_service.dart

```diff
@@ -58,4 +60,20 @@ class UserCommentRepositoryService {
 
     batch.delete(commentToDelete);
   }
+
+  /// Check if the user with id [userId] has commented at least once
+  ///  under the post with id [parentPostId].
+  Future<bool> hasUserCommentedUnderPost(
+    UserIdFirestore userId,
+    PostIdFirestore parentPostId,
+  ) async {
+    final userCommentCollection = _userCommentCollection(userId);
+
+    final userCommentQuery = await userCommentCollection
```

Comment:
```suggestion
    // only fetch one comment to check, for optimized performance
    final userCommentQuery = await userCommentCollection
```

What you did is super smart, so I though explaining it would be nice :)
Feel free to change the comment to your liking.

Reply:
Thank you :)

You are right !
This part would benefit from a little more documentation.

I add documentations in f807963a94e94ade3f22c067bd6c8768ed0c783d.

Reply:
Perfect!